### DISCUSSION TITLE
Serializer/Deserializer for the .penpot format

### DISCUSSION THREAD

**Discussion by BenJeau** [uxbox/uxbox#2308] [Ideas]
I'm looking into building some kind of converter from Adobe XD and Figma to `.penpot`. I've dug into the figma format/API and one of the last pieces that I need to figure out is how the `.penpot` file format works. I've looked into the repo, but I'm not familiar with Clojure and can't find related files to the file format.

Is there any information or script or code that can be used to serialize/deserialize to/from the .penpot format? I would like to know how to create a .penpot file from an external source.

Thanks in advance!

**Comment by niwinz** [comment]
It is better to ask it on the https://community.penpot.app/
github discusions can be considered legacy

**Comment by naltun** [comment]
> figure out is how the `.penpot` file format works

I also need to understand more details on how `.penpot` files are created and what their characteristics are. For example, I read that many applications with exportable vendor files are often just `gz`ipped JSON files but I've validated `.penpot` files are not like this.

**Reply by naltun** [reply]
@BenJeau did you ever find your answer(s)?

**Reply by BenJeau** [reply]
I've opened an issue here https://community.penpot.app/t/serializer-deserializer-for-the-penpot-format/1429 - they've at least have a figma imported/exporter and the code is opensource
